### PR TITLE
Rails Blog App: (3) Add Validations and Unit Specs

### PR DESCRIPTION
**Pull request by adarachel**
### Project requirements
Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      user = User.create(name: 'Tom')
+      post = Post.create(title: 'Hello', author: user)
+      comment = Comment.create(author: user, post:)
+
+      comment.update_post_comments_counter
+
+      expect(post.reload.comments_counter).to eq(1)
+    end
+  end
+end
```
Comment (Qoosim) [inline spec/models/comment_spec.rb:15]: - [ ] Great job so far. However, none of these tests seem not to pass. Kindly look into them and fix.

### Test Result
comment

File: spec/models/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    it 'updates the post comments_counter attribute' do
+      user = User.create(name: 'Tom')
+      post = Post.create(title: 'Hello', author: user)
+      comment = Comment.create(author: user, post:)
+
+      comment.update_post_comments_counter
+
+      expect(post.reload.comments_counter).to eq(1)
+    end
+  end
```
Comment (Qoosim) [inline spec/models/like_spec.rb:14]: - [ ] Same thing here. Kindly look at it and fix.

### Test Result

likes

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      subject.comments_counter = 'hey'
+      expect(subject).to_not be_valid
+    end
+
+    it 'comments_counter should be greater than or equal to zero' do
+      subject.comments_counter = -2
+      expect(subject).to_not be_valid
+      subject.comments_counter = 0
+      expect(subject).to be_valid
+    end
```
Comment (Qoosim) [inline spec/models/post_spec.rb:33]: - [ ] Same thing here. Kindly look into them and fix.

### Test Result
post-1


post-2

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be greater than or equal to zero' do
+      subject.posts_counter = -2
+      expect(subject).to_not be_valid
+      subject.posts_counter = 0
+      expect(subject).to be_valid
+    end
+  end
```
Comment (Qoosim) [inline spec/models/user_spec.rb:25]: - [ ] Same thing here. Kindly look into them and fix.

### Test Result

user-1


user-2

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @adarachel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by fobadara — APPROVED** [review]
Hi @adarachel, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**